packages/resolver/lib/sources/truffle/index.ts

```diff
@@ -16,10 +16,11 @@ export class Truffle implements ResolverSource {
     if (importPath === "truffle/DeployedAddresses.sol") {
       const sourceFiles = await findContracts(this.options.contracts_directory);
 
-      let abstractionFiles: string[];
-      const buildDirFiles = (abstractionFiles = fse.readdirSync(
-        this.options.contracts_build_directory
-      ));
+      let abstractionFiles: string[] =
+        fse.existsSync(this.options.contracts_build_directory)
+          ? fse.readdirSync(this.options.contracts_build_directory)
+          : [];
+      const buildDirFiles = abstractionFiles;
```

Comment:
The handoff between `abstractionFiles` and `buildDirFiles` is seems odd. Would you consider this suggestion?

```
      const buildDirFiles: string[] =
        fse.existsSync(this.options.contracts_build_directory)
          ? fse.readdirSync(this.options.contracts_build_directory)
          : [];
      const abstractionFiles = buildDirFiles.filter(file => file.match(/^.*.json$/));
```

Reply:
Sure, that makes sense.  I should note that that's preexisting; I left the existing structure in place there.  But I agree that would make sense.

(Also, is this still changes requested, or should that be gone now?)

Reply:
In any case I went ahead and did this.